### PR TITLE
Key vault secret status

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
A few minor suggestions as we discussed

File: api/secrets/models/azure-key-vault-secret-status.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package models
+
+// AzureKeyVaultSecretVersion holds a version of a Azure Key vault secret
+// swagger:model AzureKeyVaultSecretVersion
+type AzureKeyVaultSecretVersion struct {
```
[the remark this answers is not in the document]
Comment (satr) [reply api/secrets/models/azure-key-vault-secret-status.go:5]: Fixed

File: api/secrets/models/secret_status.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const (
+	// Pending In configuration but not in cluster
+	Pending SecretStatus = iota
+
+	// Consistent In configuration and in cluster
+	Consistent
+
+	// Orphan In cluster and not in configuration
+	Orphan
```
[the remark this answers is not in the document]
Comment (satr) [reply api/secrets/models/secret_status.go:14]: Fixed

File: api/secrets/models/secret.go [hunk 3]
```diff
@@ -56,3 +64,16 @@ const (
 	SecretTypeOAuth2Proxy           SecretType = "oauth2-proxy"
 	SecretTypeOrphaned              SecretType = "orphaned"
```
[the remark this answers is not in the document]
Comment (satr) [reply api/secrets/models/secret.go:65]: Removed

File: api/secrets/secret_handler.go [hunk 4]
```diff
@@ -455,49 +461,149 @@ func (eh SecretHandler) getSecretsFromComponentAuthentication(component v1.Radix
...
+	}
+	csiSecretStoreSecretMap, err := eh.getCsiSecretStoreSecretMap(envNamespace)
+	if err != nil {
+		return nil, err
+	}
 	var secrets []models.Secret
 	for _, component := range radixDeployment.Spec.Components {
-		componentSecrets, err := eh.getRadixCommonComponentSecretRefs(&component, envNamespace)
+		secretRefs := component.GetSecretRefs()
+		//activeCsiSecretProviderClassNameSet := deploymentsCsiSecretProviderNameSetMap[component.GetName()]
```
[the remark this answers is not in the document]
Comment (satr) [reply api/secrets/secret_handler.go:476]: Removed

File: api/secrets/secret_handler.go [hunk 5]
```diff
@@ -455,49 +461,149 @@ func (eh SecretHandler) getSecretsFromComponentAuthentication(component v1.Radix
...
+		componentSecrets, err := eh.getComponentSecretRefsSecrets(envNamespace, component.GetName(), &secretRefs, secretProviderClassMapForDeployment, csiSecretStoreSecretMap)
 		if err != nil {
 			return nil, err
 		}
 		secrets = append(secrets, componentSecrets...)
 	}
 	for _, jobComponent := range radixDeployment.Spec.Jobs {
-		componentSecrets, err := eh.getRadixCommonComponentSecretRefs(&jobComponent, envNamespace)
+		secretRefs := jobComponent.GetSecretRefs()
+		//activeCsiSecretProviderClassNameSet := deploymentsCsiSecretProviderNameSetMap[jobComponent.GetName()]
```
[the remark this answers is not in the document]
Comment (satr) [reply api/secrets/secret_handler.go:485]: Removed

File: api/secrets/secret_handler.go [hunk 6]
```diff
@@ -455,49 +461,149 @@ func (eh SecretHandler) getSecretsFromComponentAuthentication(component v1.Radix
...
+	secretProviderClassMap map[string]secretsstorev1.SecretProviderClass, csiSecretStoreSecretMap map[string]corev1.Secret) ([]models.Secret, error) {
+	var secrets []models.Secret
 	for _, azureKeyVault := range secretRefs.AzureKeyVaults {
+		credSecrets, err := eh.getCredentialSecretsForSecretRefsAzureKeyVault(envNamespace, componentName, azureKeyVault.Name)
+		if err != nil {
+			return nil, err
+		}
+		secrets = append(secrets, credSecrets...)
 		for _, item := range azureKeyVault.Items {
+			secretStatus := getAzureKeyVaultSecretStatus(azureKeyVault.Name, secretProviderClassMap, csiSecretStoreSecretMap)
```
[the remark this answers is not in the document]
Comment (satr) [reply api/secrets/secret_handler.go]: Moved

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**